### DISCUSSION TITLE
How to handle thread-locking code?

### DISCUSSION THREAD

**Discussion by BrandonConder** [fonsp/Pluto.jl#858] [Questions about using Pluto]
When I started with Pluto, I managed to lose my work on my first notebook while I was experimenting with multithreading†.

**Is there a feature to open a notebook for editing without executing it?**

I've since determined that you can (as far as I can tell) ruin any notebook by adding a cell with the code `while true; end`, making sure it's saved (which might require running and interrupting the cell?), and closing/shutting down the notebook.

From then on, whenever I open such a notebook it locks up Pluto entirely. The web browser hangs up like it's trying to load a page, but nothing ever shows up. Even using `Ctrl+C` at the REPL does nothing.

I've tried this with Julia 1.5.3x64 on Ubuntu , Windows 10, and Ubuntu under Windows Subsystem Linux 2 (WSL2) and the behavior is the same (with the exception that the Windows binary can't interrupt cells).

As a novice with Pluto, as far as I can tell this makes the notebook unusable. I'm not sure if it's possible to remove the offending cell without rewriting the hashes, so my only option was to rebuild the notebook using the text within the notebook file.

I've attached the simplest example here: [myworstnightmare.zip](https://github.com/fonsp/Pluto.jl/files/5826629/myworstnightmare.zip)

```jl
### A Pluto.jl notebook ###
# v0.12.18

using Markdown
using InteractiveUtils

# ╔═╡ 94358b7a-58f4-11eb-03aa-35782c247258
while true; end

# ╔═╡ Cell order:
# ╠═94358b7a-58f4-11eb-03aa-35782c247258
```

So, what's the recommended way to handle thread-locking code?

1. Are there ways to open notebooks for editing without executing? If so can it be made obvious for new users?

2. Is it possible (perhaps only if julia has more than one thread) to have the Pluto code run periodic interrupts on a notebook to ensure that at least the notebook page is served up and `Ctrl+C` can be serviced?

† I did eventually figure out that in this specific case, it was possible to interrupt the cell and yield to the main process, but by then I already corrupted the notebook by trying to edit out the offending cell with a text editor. Also note that cells can't be interrupted from the Windows Julia binary.

**Comment by fonsp** [comment]
Hey @BrandonConder !

Eventually we will implement #259 to fix this, right now your options are:
- Edit the notebook in a text editor to remove the blocking code, or
- Start Pluto with `Pluto.run(run_notebooks_on_load=false)` (or something similar). But **be careful**: this can lead to hidden state problems, exactly what we are trying to avoid! We might remove this option in the future.
